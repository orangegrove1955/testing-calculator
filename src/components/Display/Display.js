import React from "react";
import PropTypes from "prop-types";

const Display = ({ displayValue }) => {
  return (
    <div className="display-container">
      <p className="display-value">{displayValue}</p>
    </div>
  );
};

Display.propTypes = { displayValue: PropTypes.number.isRequired };

export default Display;
