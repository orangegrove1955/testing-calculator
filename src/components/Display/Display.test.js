import React from "react";
import { shallow, render } from "enzyme";
import Display from "../Display/Display";

describe("Calculator component", () => {
  let wrapper;

  beforeEach(() => (wrapper = shallow(<Display displayValue={0} />)));

  it("should render a <div />", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("should render a display value in a p", () => {
    let deepWrapper = render(<Display displayValue={0} />);
    const displayP = deepWrapper.find(".display-value");
    expect(displayP.length).toEqual(1);
  });

  it("correctly renders 0 value of displayValue", () => {
    wrapper.setProps({ displayValue: 0 });
    expect(wrapper.text()).toEqual("0");
  });

  it("correctly renders positive value of displayValue", () => {
    wrapper.setProps({ displayValue: 10 });
    expect(wrapper.text()).toEqual("10");
  });

  it("correctly renders negative value of displayValue", () => {
    wrapper.setProps({ displayValue: -10 });
    expect(wrapper.text()).toEqual("-10");
  });
});
