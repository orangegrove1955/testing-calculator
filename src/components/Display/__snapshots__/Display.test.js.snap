// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component should match Display snapshot render 1`] = `ShallowWrapper {}`;
